tools/autograd/derivatives.yaml

```diff
@@ -830,6 +830,17 @@
   self: at::_unsafe_index_put(grad.new_zeros_symint(self.sym_sizes(), self.options()), indices, grad, true)
   result: auto_linear
 
+- name: _unsafe_masked_index(Tensor self, Tensor mask, Tensor?[] indices, Scalar fill) -> Tensor
+  self: at::_unsafe_masked_index_put(grad.new_zeros_symint(self.sym_sizes(), self.options()), mask, indices, grad, true)
+  mask: non_differentiable
+  result: _unsafe_masked_index(self_t, mask, indices, 0)
+
+- name: _unsafe_masked_index_put(Tensor self, Tensor mask, Tensor?[] indices, Tensor values, bool accumulate=False) -> Tensor
+  self: "accumulate ? grad.masked_fill(~mask, 0) : at::_unsafe_masked_index_put(grad, mask, indices, zeros_like(values), false)"
```

Comment:
When `accumulate=True` why do you zero out the self gradient? `self` should be included in the accumulated result and therefore have equal gradient.

Reply:
You are right. This hasn't been caught by CI because we don't have (yet) an OpInfo for this op, and we don't really use it. But it'd be good to have for completeness.

@isuruf change `grad.masked_fill(~mask, 0)` by `grad` and we should be good here.